apps/language_server/lib/language_server/experimental/provider/code_action/replace_remote_function.ex

```diff
@@ -0,0 +1,147 @@
+      |> List.pop_at(-1)
+
+    {module, name}
+  end
+
+  @suggestion_pattern ~r/\* .*\/[\d]+/
+  defp extract_suggestions(message) do
+    case Regex.scan(@pattern, message) do
+      [[_, _, arity, suggestions_string]] ->
+        suggestions =
+          @suggestion_pattern
+          |> Regex.scan(suggestions_string)
+          |> Enum.flat_map(fn [suggestion] ->
+            case String.split(suggestion, [" ", "/"]) do
+              ["*", name, ^arity] -> [String.to_atom(name)]
+              _ -> []
+            end
+          end)
+
+        {:ok, suggestions}
+
+      _ ->
+        :error
+    end
+  end
+
+  defp extract_line(%Diagnostic{} = diagnostic) do
+    diagnostic.range.start.line
+  end
+
+  defp build_code_actions(%SourceFile{} = source_file, one_based_line, module, name, suggestions) do
+    with {:ok, line_text} <- SourceFile.fetch_text_at(source_file, one_based_line),
+         {:ok, line_ast} <- Ast.from(line_text),
+         {:ok, edits_per_suggestion} <-
+           text_edits_per_suggestion(line_text, line_ast, module, name, suggestions) do
+      case edits_per_suggestion do
+        [] ->
+          :error
+
+        [_ | _] ->
+          edits_per_suggestion =
+            Enum.map(edits_per_suggestion, fn {text_edits, suggestion} ->
+              text_edits = Enum.map(text_edits, &update_line(&1, one_based_line))
+              {text_edits, suggestion}
+            end)
+
+          replies =
+            Enum.map(edits_per_suggestion, fn {text_edits, function_name} ->
+              CodeActionResult.new(
+                title: construct_title(module, function_name),
+                kind: :quick_fix,
+                edit: Workspace.Edit.new(changes: %{source_file.uri => text_edits})
+              )
+            end)
+
+          {:ok, replies}
+      end
+    end
+  end
+
+  defp text_edits_per_suggestion(line_text, line_ast, module, name, suggestions) do
+    Enum.reduce(suggestions, {:ok, []}, fn
+      suggestion, {:ok, edits_per_suggestions} ->
+        case CodeMod.ReplaceRemoteFunction.text_edits(
+               line_text,
+               line_ast,
+               module,
+               name,
+               suggestion
+             ) do
+          {:ok, []} -> {:ok, edits_per_suggestions}
+          {:ok, text_edits} -> {:ok, [{text_edits, suggestion} | edits_per_suggestions]}
+          :error -> :error
+        end
+
+      _suggestion, :error ->
+        :error
+    end)
+  end
+
+  defp update_line(%TextEdit{} = text_edit, line_number) do
+    text_edit
+    |> put_in([:range, :start, :line], line_number - 1)
+    |> put_in([:range, :end, :line], line_number - 1)
+  end
+
+  defp construct_title(module_list, function_name) do
+    module_string =
+      module_list
+      |> Enum.map(fn module ->
+        module
+        |> Atom.to_string()
+        |> String.trim_leading("Elixir.")
+      end)
+      |> Enum.join(".")
+
+    "Replace function with #{module_string}.#{function_name}"
```

Comment:
this is going to be pretty long in editors. I'd suggest dropping "function" and make it "Replace with"